### PR TITLE
[PDS]: Setup backup targets w.r.t supported Object stores.

### PR DESCRIPTION
**Pull request by gauraa2**
This is the first backup PR, as part of this PR following packages and tests are added.
- pdsbackup package
      - This package comprises of pds backup utilities , which include various functions to create/delete backup creds/targets.
      - Also this comprises of helper functions for creating/deleting cloud object storage(includes AWS, GCP, Azure and S3 compatible)
 
-  Added backup test to setup backup targets w.r.t all the supported Object stores.

Logs: http://aetos.pwx.purestorage.com/resultSet/testSetID/143150

### CODE REVIEW

**Comment by Madanagopal19** [comment]
Attach jenkins logs and add description about this testcase

File: tests/pds/backup_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+var _ = Describe("{ValidateBackupTargetsOnSupportedObjectStores}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("ValidateBackupTargetsOnSupportedObjectStores", "Validate backup targets for all supported object stores.", nil, 0)
+		bkpClient, err = pdsbkp.InitializePdsBackup()
+		log.FailOnError(err, "Failed to initialize backup for pds.")
+	})
+
+	It("add all supported Object stores as backup target for data services", func() {
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:28]: nit: Add all supported Object stores as backup target for data services

File: tests/pds/backup_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		Step("Create AWS S3 Backup target.", func() {
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:29]: Add step log to aetos dash board

File: tests/pds/backup_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+var (
+	bkpClient                                                             *pdsbkp.BackupClient
+	awsBkpTargets, azureBkpTargets, s3CompatibleBkpTargets, gcpBkpTargets []*pds.ModelsBackupTarget
+)
+
+var _ = Describe("{ValidateBackupTargetsOnSupportedObjectStores}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("ValidateBackupTargetsOnSupportedObjectStores", "Validate backup targets for all supported object stores.", nil, 0)
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:23]: add pdsLabels

File: tests/pds/backup_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+			s3CompatibleBkpTargets = append(s3CompatibleBkpTargets, bkpTarget)
+		})
+	})
+	JustAfterEach(func() {
+		deleteAllBkpTargets()
+	})
+})
+
+func deleteAllBkpTargets() {
+	log.Info("Delete all the backup targets.")
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:60]: use log.InfoD

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]
Minor comment

File: tests/pds/backup_test.go (hunk 2 above)
Comment (gauraa2) [reply tests/pds/backup_test.go]: Done

File: tests/pds/backup_test.go (hunk 3 above)
Comment (gauraa2) [reply tests/pds/backup_test.go]: Done

File: tests/pds/backup_test.go (hunk 4 above)
Comment (gauraa2) [reply tests/pds/backup_test.go]: Done

File: tests/pds/backup_test.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+			log.InfoD(stepLog)
+			bkpTarget, err := bkpClient.CreateAwsS3BackupCredsAndTarget(tenantID, fmt.Sprintf("%v-aws", bkpTargetName))
+			log.FailOnError(err, "Failed to create AWS backup target.")
+			log.InfoD("AWS S3 target - %v created successfully", bkpTarget.GetName())
+			awsBkpTargets = append(awsBkpTargets, bkpTarget)
+		})
+		stepLog = "Create GCP Backup target."
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			bkpTarget, err := bkpClient.CreateGcpBackupCredsAndTarget(tenantID, fmt.Sprintf("%v-gcp", bkpTargetName))
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:40]: how are we validating the backup target is created successfully or not ?
say if its api call we need to we need to assert its response, is that assertion part handled?

File: tests/pds/backup_test.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			bkpTarget, err := bkpClient.CreateS3CompatibleBackupCredsAndTarget(tenantID, fmt.Sprintf("%v-s3compatible", bkpTargetName))
+			log.FailOnError(err, "Failed to create S3 compatible backup target.")
+			log.InfoD("S3 compatible backup target - %v created successfully", bkpTarget.GetName())
+			s3CompatibleBkpTargets = append(s3CompatibleBkpTargets, bkpTarget)
+		})
+	})
+	JustAfterEach(func() {
+		deleteAllBkpTargets()
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:63]: similarly for delete backup targets, we need to validate if all the backup targets are deleted successfully.

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**